src/lib.rs

```diff
@@ -129,54 +129,54 @@ where
     }
     if let Some(build_id) = object.build_id()? {
         let path = locate_debug_build_id(build_id);
-        if path.is_ok() {
-            return path;
+        if let Some(path) = path {
```

Comment:
You could chain this with the `build_id` call if you wanted to avoid nesting conditionals like this:

```rust
if let Some(path) = object.build_id()?.and_then(|build_id| locate_debug_build_id(build_id)) {
  return Ok(Some(path));
}
```

Reply:
Love it, will do!